x-pack/legacy/plugins/siem/public/containers/case/use_get_cases.tsx

```diff
@@ -142,5 +184,73 @@ export const useGetCases = (): [
       didCancel = true;
     };
   }, [state.queryParams, state.filterOptions]);
-  return [state, setQueryParams, setFilters];
+  useEffect(() => fetchCases(), [state.queryParams, state.filterOptions]);
+
+  const getCaseCount = useCallback((caseState: keyof CaseCount) => {
+    let didCancel = false;
+    const fetchData = async () => {
+      dispatch({ type: 'FETCH_INIT', payload: 'caseCount' });
+      try {
+        const response = await getCases({
+          filterOptions: { search: '', state: caseState, tags: [] },
+        });
+        if (!didCancel) {
+          dispatch({
+            type: 'FETCH_CASE_COUNT_SUCCESS',
+            payload: { [caseState]: response.total },
+          });
+        }
+      } catch (error) {
+        if (!didCancel) {
+          errorToToaster({ title: i18n.ERROR_TITLE, error, dispatchToaster });
+          dispatch({ type: 'FETCH_FAILURE', payload: 'caseCount' });
+        }
+      }
+    };
+    fetchData();
+    return () => {
+      didCancel = true;
+    };
+  }, []);
+
+  const dispatchUpdateCaseProperty = useCallback(
+    ({ updateKey, updateValue, caseId, version }: UpdateCase) => {
+      let didCancel = false;
+      const fetchData = async () => {
+        dispatch({ type: 'FETCH_INIT', payload: 'caseUpdate' });
+        try {
+          await updateCaseProperty(
+            caseId,
+            { [updateKey]: updateValue },
+            version ?? '' // saved object versions are typed as string | undefined, hope that's not true
+          );
+          if (!didCancel) {
+            dispatch({ type: 'FETCH_UPDATE_CASE_SUCCESS' });
+            fetchCases();
+            getCaseCount('open');
+            getCaseCount('closed');
+          }
+        } catch (error) {
+          if (!didCancel) {
+            errorToToaster({ title: i18n.ERROR_TITLE, error, dispatchToaster });
+            dispatch({ type: 'FETCH_FAILURE', payload: 'caseUpdate' });
+          }
+        }
+      };
+      fetchData();
+      return () => {
+        didCancel = true;
+      };
+    },
+    [filterQuery, state.filterOptions]
```

Comment:
@XavierM this is weird, if I dont include these dependencies, when i call `fetchCases()` from this function it calls it with initialState filterOptions, not updated. i wouldnt expect this since `fetchCases` has filterOptions as a dependency?

Reply:
Yes I think that I have a fix in my PR for it, so let's move one

Reply:
look at this commit in my pr coming https://github.com/elastic/kibana/pull/59265/commits/ff92b62054bef5dbc85e005aab19d85ccff383ad